database.c

```diff
@@ -728,6 +737,65 @@ database_save_later(int secs)
   }
 }
 
+void
+migrate_configs(const char *legacydir, const char *currentdir)
+{
+  DIR *dptr;
+  struct dirent *dir;
+  FILE *src, *dest;
+  char srcname[260], destname[260];
+  int ch;
+  errno = 0;
+  if (rename(legacydir, currentdir) < 0)
+  {
+    log_warning("Rename failed with %s. Falling back to copy/delete.\n", strerror(errno));
+    if (mkdir(currentdir, 0755) < 0 && errno != EEXIST)
+    {
+      log_err("%s: mkdir: %s\n", currentdir, strerror(errno));
+      return;
+    }
+    dptr = opendir(legacydir);
+    if (dptr)
+    {
+      while ((dir = readdir(dptr)) != NULL)
+      {
+        if (dir->d_type == DT_REG)
+        {
+          snprintf(srcname, sizeof(srcname), "%s/%s", legacydir, dir->d_name);
+          snprintf(destname, sizeof(destname), "%s/%s", currentdir, dir->d_name);
```

Comment:
In general: 'sizeof' is an operator, not a function -> use "sizeof OBJ", not "sizeof(OBJ)" [and yes, sometimes they are needed and sizeof (TYPE) is different thing etc but still].